tests/functional/test_logger_powertools_formatter.py

```diff
@@ -288,3 +290,22 @@ def test_log_formatting(stdout, service_name):
 
     # THEN the formatting should be applied (NB. this is valid json, but hasn't be parsed)
     assert log_dict["message"] == '["foo bar 123 [1, None]", null]'
+
+
+def test_log_json_indent_compact_indent(stdout, service_name, monkeypatch):
+    # GIVEN a logger with default settings and WHEN POWERTOOLS_DEV is not set
+    if "POWERTOOLS_DEV" in os.environ:
+        monkeypatch.delenv(name="POWERTOOLS_DEV")
```

Comment:
```suggestion
    monkeypatch.delenv(name="POWERTOOLS_DEV", raising=False)
```

[`delenv`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.MonkeyPatch.delenv) supports a flag to not raise an exception if the env var doesn't exist.